xrpl/utils/txn_parser/balance_changes.py

```diff
@@ -0,0 +1,137 @@
+"""
+Parse balance changes, final balances and previous balances of every
+account involved in the given transaction.
+"""
+from __future__ import annotations
+
+from decimal import Decimal
+from typing import Any, Dict, List, Union, cast
+
+from xrpl.utils.txn_parser.utils import (
+    RawTxnType,
+    SubscriptionRawTxnType,
+    compute_balance_changes,
+    normalize_nodes,
+    normalize_transaction,
+    parse_final_balance,
+    parse_quantities,
+    validate_transaction_fields,
+)
+
+
+def parse_previous_balances(
+    transaction: Union[RawTxnType, SubscriptionRawTxnType],
+) -> Dict[str, Any]:
+    """Parse the previous balances of all accounts affected
+    by the transaction before it occurred.
+
+    Args:
+        transaction (Union[RawTxnType, SubscriptionRawTxnType]):
+            Raw transaction data including the account that
+            sent the transaction and the affected nodes.
+
+    Returns:
+        Dict[str, Any]:
+            All previous balances.
```

Comment:
nit: this should just be
```suggestion
    Returns:
        All previous balances.
```
ditto with the other docstrings